### PR TITLE
Add documentation for lsp-mode, another Emacs client

### PR DESCRIPTION
**Pull request by danielmartin**
This PR adds some documentation for `lsp-mode`, another Emacs client that
supports Clangd, including some of its private extensions to Language Server Protocol.

### CODE REVIEW

File: installation.md [hunk 1]
```diff
@@ -152,11 +152,55 @@ with `M-x company-mode`.
...
+### Find workspace symbols
+
+You can search for workspace symbols using a package like [lsp-ivy](https://github.com/emacs-lsp/lsp-ivy) or [helm-lsp](https://github.com/emacs-lsp/helm-lsp)
+
+![Find workspace symbols in lsp-mode](screenshots/lsp_mode_workspace_symbols.png)
+
+### Additional UI features
+
+Additional UI features like peek references, peek definition, etc. are available as a separate package, [lsp-ui](https://github.com/emacs-lsp/lsp-ui)
 </details>
```
Comment (sam-mccall) [inline installation.md:199]: we need to have the "under the hood" section briefly explaining how to get to clangd logs and how to set clangd flags/binary path. Other parts of the docs (and the bug tracker) assume the user knows how to do this in their editor.

File: installation.md [hunk 2]
```diff
@@ -152,11 +152,55 @@ with `M-x company-mode`.
...
+
+Add the following to `~/.emacs` to enable lsp-mode:
+
+```
+(require 'lsp-mode)
+(add-hook 'c-mode-hook #'lsp)
+(add-hook 'c++-mode-hook #'lsp)
+```
+
+### Indexing status
```
Comment (sam-mccall) [inline installation.md:170]: This is nice but it's not really a feature needed to get started and the user should discover it without any action needed (if i'm understanding correctly). So I'm not sure we need to mention this.

File: installation.md [hunk 3]
```diff
@@ -152,11 +152,55 @@ with `M-x company-mode`.
...
+
+lsp-mode can process indexing notifications from clangd and show you progress in the mode line.
+
+![Indexing status in lsp-mode](screenshots/lsp_mode_indexing.png)
+
+### Clang-tidy error explanations
```
Comment (sam-mccall) [inline installation.md:176]: These *are* worth mentioning as there's some integrations with other plugins/commands that users need to know.

But having full headings and screenshots for each is too much weight, as these are all relatively minor features (e.g. for the other editors we just show a single screenshot with completion and/or diagnostics, which are probably the most prominent features).

I'd suggest adding a bullet list instead:

```
### Integration with other packages:

 - See documentation for clang-tidy linting issues with `M-x flycheck-explain-error-at-point`.
 - Search for workspace symbols using [lsp-ivy] or [helm-lsp]
 - [lsp-ui] provides extra UI features: ...
```

File: installation.md [hunk 4]
```diff
@@ -152,11 +152,55 @@ with `M-x company-mode`.
...
+
+The clangd Flycheck integration in lsp-mode shows error explanations from the LLVM website for clang-tidy linting issues.
+This is very useful when you need context before applying a clang-tidy fix. Press `M-x flycheck-explain-error-at-point` when the point is at a clang-tidy error.
+
+![clang-tidy error explanations in lsp-mode](screenshots/lsp_mode_clang_tidy_explanations.png)
+
+### Type signatures in mode line
+
+When the point is at a symbol, lsp-mode extracts a function prototype and shows it in the mode line. Simply placing the
```
Comment (sam-mccall) [inline installation.md:185]: the wording is a bit confusing here: function prototype surely only if it's a function :-)
Maybe "When the point is at a symbol, its declaration is shown in the mode line". Again I'm not sure we need an extra screenshot for this.

File: installation.md (hunk 1 above)
Comment (danielmartin) [reply installation.md:187]: Done.

File: installation.md (hunk 2 above)
Comment (danielmartin) [reply installation.md]: I've removed it.

File: installation.md (hunk 3 above)
Comment (danielmartin) [reply installation.md]: Good point. I've added a summary of the integrations `lsp-mode` has with other packages.

File: installation.md (hunk 4 above)
Comment (danielmartin) [reply installation.md]: Given the rest of the suggestions, there's no point in mentioning this feature, so I've remove the section.

**Pull request closed without merging**